### PR TITLE
[move] Enable documentation generator 

### PR DESCRIPTION
**Pull request by wrwg**
This PR enables the Move documentation generator for Aptos. The generator is available via the cli:

```
# aptos move document --help
aptos-move-document 1.0.1
Documents a Move package

This converts the content of the package into markdown for documentation.

USAGE:
    aptos move document [OPTIONS]

OPTIONS:
        --collapsed-sections
            Whether details should be put into collapsed sections. This is not supported by all
            markdown, but the github dialect

        --include-impl
            Whether to include private declarations and implementations into the generated
            documentation

        --include-specs
            Whether to include specifications in the generated documentation

          --specs-inlined
            Whether specifications should be put side-by-side with declarations or into a separate
            section

            <move package options>

```

It's also integrated into the framework building process and will be automatically triggered similar as other derived artifacts, via `cached-packages` or explicit release building.

The generated documentation is reachable via github, here is an [example created by this PR](https://github.com/wrwg/aptos-core/blob/docgen/aptos-move/framework/aptos-framework/doc/account.md). We likely need to massage the move sources to get optimal appearance.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/5187)

### CODE REVIEW

**Review by alinush — APPROVED** [review]
LGTM!

**Review by clay-aptos — APPROVED** [review]
Wow! Thank you so much, Wolfgang! I will help get this linked up.

**Comment by rajkaramchedu** [comment]
I like that this is now part of the CLI. Purely for recording my understanding of this feature, I am including the following:
- The resulting Markdown output is not ready for Docusaurus (as is already noted indirectly in this PR description). In specific, the `<a name="0x1_account"></a>` syntax will not be parsed by Docusaurus html generator, it will give out error and stop.
- I am not entirely sure if the `<dl>...</dt>` syntax passes Docusaurus parser. 
In any case, love this! Danke!

**Comment by wrwg** [comment]
> I like that this is now part of the CLI. Purely for recording my understanding of this feature, I am including the following:
> 
> * The resulting Markdown output is not ready for Docusaurus (as is already noted indirectly in this PR description). In specific, the `<a name="0x1_account"></a>` syntax will not be parsed by Docusaurus html generator, it will give out error and stop.
> * I am not entirely sure if the `<dl>...</dt>` syntax passes Docusaurus parser.
>   In any case, love this! Danke!

I'm kind of ignorant regards html current state, so when I create this tool two years ago, I choose just whatever worked on github. If you tell me how I should fix what in the output, happy to do this!

**Comment by rajkaramchedu** [comment]
> > I like that this is now part of the CLI. Purely for recording my understanding of this feature, I am including the following:
> > 
> > * The resulting Markdown output is not ready for Docusaurus (as is already noted indirectly in this PR description). In specific, the `<a name="0x1_account"></a>` syntax will not be parsed by Docusaurus html generator, it will give out error and stop.
> > * I am not entirely sure if the `<dl>...</dt>` syntax passes Docusaurus parser.
> >   In any case, love this! Danke!
> 
> I'm kind of ignorant regards html current state, so when I create this tool two years ago, I choose just whatever worked on github. If you tell me how I should fix what in the output, happy to do this!

My thinking is we should not prioritize for Docusaurus for the time being. With this PR, the CLI is ready to use to generate a GitHub-hostable MD, and any doc in Docusaurus will link to it. Good enough for now, I think. We should move on to bigger and more important things for the docs.

**Comment by rajkaramchedu** [comment]
I am saying this because we are better off spending time on writing fresh new Move examples for various use cases like contracts, dapps or whatever we decide is important for a newbie Move developer. These will be, because we will create them new by hand, hosted on Docusaurus. Better value for our user, no?

**Review by gregnazario — APPROVED** [review]

File: aptos-move/framework/src/docgen.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#[derive(Debug, Clone, clap::Parser, serde::Serialize, serde::Deserialize, Default)]
+pub struct DocgenOptions {
+    /// Whether to include private declarations and implementations into the generated
+    /// documentation.
+    #[clap(long)]
+    pub include_impl: bool,
+
+    /// Whether to include specifications in the generated documentation.
+    #[clap(long, global = true)]
+    pub include_specs: bool,
```
[the remark this answers is not in the document]
Comment (gregnazario) [reply aptos-move/framework/src/docgen.rs:20]: Resolved, it stays within scope I checked

File: crates/aptos/src/move_tool/mod.rs [hunk 2]
```diff
@@ -97,6 +99,7 @@ impl MoveTool {
             MoveTool::RunScript(tool) => tool.execute_serialized().await,
             MoveTool::Test(tool) => tool.execute_serialized().await,
             MoveTool::Prove(tool) => tool.execute_serialized().await,
+            MoveTool::Document(tool) => tool.execute_serialized().await,
```
[the remark this answers is not in the document]
Comment (gregnazario) [reply crates/aptos/src/move_tool/mod.rs:102]: Up to you, I'm fine either way.

[the remark this answers is not in the document]
Comment (wrwg) [reply crates/aptos/src/move_tool/mod.rs:102]: I had the `doc` first, then `document` just looked better ;-)

**Comment by alnoki** [comment]
Addresses #2142

**Pull request merged**